components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtGoogleAnalyticsTrackingHandler.java

```diff
@@ -67,6 +72,14 @@
 
 	@Override
 	public boolean handleRequest(MessageContext msgCtx) {
+        TracingSpan span = null;
+        Boolean tracingEnabled = Boolean.valueOf(ServiceReferenceHolder.getInstance().getAPIManagerConfiguration()
```

Comment:
Instead of loading the APIM configuration for each invocation, you can check whether tracingEnabled during the startup and set the boolean value during the handler initialization. And then read the variable.

Reply:
fixed